### PR TITLE
UI: Implement querying/updating/creating ingredients/meals/recipes

### PR DESCRIPTION
**Pull request by feds01**

### CODE REVIEW

File: src/components/MealTable/index.tsx [hunk 1]
```diff
@@ -32,11 +33,19 @@ export const MealLink = ({ name, id }: MealLinkProps) => {
 };
 
 type MealTableProps = {
```
Comment (kontheocharis) [inline src/components/MealTable/index.tsx:35]: Can use interface

File: src/components/MealTable/index.tsx [hunk 2]
```diff
@@ -32,11 +33,19 @@ export const MealLink = ({ name, id }: MealLinkProps) => {
...
   items: Meal[];
-  pageSize?: number;
+  // Current pagination state.
+  pagination: Pagination;
+  // Function that handles pagination changes from the table
+  onPaginationChange: (value: Pagination) => void;
 };
 
-const MealTable = ({ items, pageSize = 20 }: MealTableProps) => {
+const MealTable = ({
```
Comment (kontheocharis) [inline src/components/MealTable/index.tsx:44]: Inconsistency between usage of arrow functions and `function` for components

File: src/forms/MutateRecipe.tsx [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+                      <MenuItem key={unit} value={unit}>
+                        {unit}
+                      </MenuItem>
+                    ))}
+                  </TextField>
+                </Box>
+                <IconButton onClick={() => remove(index)} sx={{ mb: 2.5 }}>
+                  <MdOutlineDelete color="error" />
+                </IconButton>
+              </Box>
```
Comment (kontheocharis) [inline src/forms/MutateRecipe.tsx:199]: Maybe factor this out into a component?

**Review by kontheocharis — CHANGES_REQUESTED** [review]

File: src/forms/MutateRecipe.tsx [hunk 4]
```diff
@@ -19,7 +24,120 @@ import { trpc } from "../utils/trpc";
 import { MdOutlineDelete } from "react-icons/md";
 import { useCallback, useState } from "react";
 
-type CreateRecipeFormProps = {
+// @@Todo: move this into the backend, and this should be narrowed down by the
+// unit type that is specified on the ingredient.
+const UNITS = ["piece", "Kg", "g", "ml", "l"];
```
Comment (kontheocharis) [inline src/forms/MutateRecipe.tsx:29]: Kg -> kg

File: src/forms/MutateRecipe.tsx [hunk 5]
```diff
@@ -19,7 +24,120 @@ import { trpc } from "../utils/trpc";
 import { MdOutlineDelete } from "react-icons/md";
 import { useCallback, useState } from "react";
 
-type CreateRecipeFormProps = {
+// @@Todo: move this into the backend, and this should be narrowed down by the
+// unit type that is specified on the ingredient.
```
Comment (kontheocharis) [inline src/forms/MutateRecipe.tsx:28]: Move what in backend?

Comment (feds01) [reply src/forms/MutateRecipe.tsx:28]: Just this list so it is available for both the backend and frontend without duplicating the definition

**Review by kontheocharis — APPROVED** [review]

**Pull request merged**